src/goto-analyzer/get_function_pointer_values.cpp

```diff
@@ -0,0 +1,113 @@
+/*******************************************************************\
+
+Module: Show function pointer values
+
+Author: Diffblue Ltd, 2020
+
+\*******************************************************************/
+
+#include "get_function_pointer_values.h"
+
+#include <util/message.h>
+#include <util/namespace.h>
+
+#include <goto-programs/goto_model.h>
+#include <goto-programs/restrict_function_pointers.h>
+
+#include <analyses/ai.h>
+#include <analyses/variable-sensitivity/abstract_enviroment.h>
+#include <analyses/variable-sensitivity/value_set_abstract_object.h>
+#include <analyses/variable-sensitivity/variable_sensitivity_domain.h>
+
+#include <regex>
+#include <string>
+
+void get_function_pointer_values(
+  const goto_modelt &goto_model,
+  const ai_baset &ai,
+  const optionst &options,
+  message_handlert &message_handler,
+  std::string filename)
+{
```

Comment:
I would strongly like (perhaps in an immediate follow up PR) this function to be greatly simplified: having 6 levels of nesting and a nearly 100 line function are both code smells. At a minimum, each loop could be pulled out into a function that deals with one iteration